lib/insights/api/common/status.rb

```diff
@@ -0,0 +1,50 @@
+module Insights
+  module API
+    module Common
+      module Status
+        module Api
+          def health
+            @output = {}
+
+            if db_up?
+              @output[:status] = migration_context.needs_migration? ? "warn" : "pass"
+              @output[:database_status] = "up"
+            else
+              @output[:status] = "fail"
+              @output[:database_status] = "down"
+            end
+
+            @output[:pending_migrations] = pending_migrations
+
+            render :json => @output, :status => status
+          end
+
+          private
+
+          def migration_context
+            ActiveRecord::Base.connection.migration_context
+          end
+
+          def pending_migrations
+            migration_context.migrations_status.collect(&:first).count("down")
```

Comment:
If the `entrypoint` (or `container.spec.args -c ...`) was changed to `bundle exec rake db:migrate && bundle exec rails s`
This `pending_migrations` check would not be needed.

However if the container startup command is not changed then I would like to keep this here as we've already ( in Catalog ) experienced deployments that deploy but are not usable due to pending migrations.

Reply:
I'm not sure I understand this use case.  What handles migrating and seeding?

Reply:
The main use-case here is that we (as in Catalog) have had a few deployments appear to go smoothly, but then a piece of functionality is broken (icons several times in our case) due to a failed migration. 

This would just add a little piece of troubleshooting just in case a migration failed and the application _appears_ to be running normally, but something is degraded somehow.

Reply:
Shouldn't the new api container fail to start if the migration fails?

Reply:
Currently it doesn't fail - we could change that though if we changed the entrypoint for catalog/others to `bundle exec rake db:migrate && bundle exec rails server`, but it just continues currently, as seen here:
https://github.com/RedHatInsights/catalog-api/blob/f1fb5bf0b288c4543029926780f97490bf0219aa/docker-assets/run_rails_server#L27-L28

Is that a change we'd be comfortable with doing? We could remove the migration checking stuff from this if we did it. @syncrou @carbonin @gmcculloug

Reply:
@lindgrenj6 Let's move the migration discussion to https://github.com/RedHatInsights/insights-api-common-rails/issues/146